File: Smallest number repeating K times/Smallest_number_repeating_K_times.cpp
class Solution {
  public:
    int findDuplicate(const vector<int>& arr, int k) {
        unordered_map<int,int> mpp;
        for(auto i: arr) {
            mpp[i]++;
        }
        int ans = 1e9;
        for(auto i: mpp) {
            if(i.second == k && ans > i.first) {
                ans = i.first;
            }
        }
        if(ans == 1e9) return -1;
        return ans;
    }
};

File: Smallest number repeating K times/Smallest_number_repeating_K_times.py
class Solution:
    def findDuplicate(self, arr, k):
        mpp = {}
        for i in arr:
            mpp[i] = mpp.get(i, 0) + 1
        ans = float('inf')
        for key, value in mpp.items():
            if value == k and ans > key:
                ans = key
        return -1 if ans == float('inf') else ans

